### PR TITLE
migrate partition-values tests to TS

### PR DESCRIPTION
**Pull request by JacobOscarGunnarsson**

### CODE REVIEW

**Review by takameyer — COMMENTED** [review]
In general, it would be great if we could somehow come up with a way to not have to deploy a new app for every partition value type.  Might be a good refactor later one to reduce the amount of apps being deployed.

File: integration-tests/tests/src/tests/sync/partition-values.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+      const realmConfigPrimary = createConfig(PvIntDog, this.user, 42);
+      const realmConfigSecondary = createConfig(PvIntDog, this.user, 43);
+
+      console.log(realmConfigPrimary);
+      // ensure clean starting point
+      Realm.deleteFile(realmConfigPrimary);
+
+      const realm1 = await Realm.open(realmConfigPrimary);
+      realm1.write(() => {
+        realm1.create("Dog", { _id: new ObjectId(), name: "King" });
```
Comment (takameyer) [inline integration-tests/tests/src/tests/sync/partition-values.ts:91]: the `realm_id` is not set (which is the primary key).  I'm kind of surprised this works at all.
Just did a bit of reading.  I think in general, when the configuration is opened, it just applies a partition to the arbitrary value 42, and any data written there will be partitioned by this, without actually having to explicitly have it's object level field (in this case `realm_id`) to anything at all.
Seems like these tests don't really reflect how most users will use partition based sync, but we are trying to migrate everyone away from that anyway.
Nothing to do here, just was confused what was happening here.

**Review by takameyer — APPROVED** [review]
Just some small things.  
Very strange tests.  Will be great to delete them when Flexible Sync reaches 100% adoption.

File: integration-tests/realm-apps/pv-int-tests/config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "pv-int-tests",
+    "sync": {
+        "development_mode_enabled": false
```
[the remark this answers is not in the document]
Comment (takameyer) [reply integration-tests/realm-apps/pv-int-tests/config.json:4]: In any case, after reading the tests, we can probably just leave this alone.

**Review by kraenhansen — CHANGES_REQUESTED** [review]
Since this PR isn't making changes to any `package.json` files, I'd say the changes to the `package-lock.json` file should be left out. Feel free to open a separate PR with the changes if those changes are actually needed.

**Review by kraenhansen — APPROVED** [review]

**Pull request merged**